pkgs/tools/misc/immich-cli/default.nix

```diff
@@ -1,31 +1,53 @@
 { lib
 , buildNpmPackage
 , fetchFromGitHub
+, testers
 }:
 
-buildNpmPackage {
-  pname = "immich-cli";
-  version = "2.0.6";
-
+let
+  version = "2.2.4";
   src = fetchFromGitHub {
     owner = "immich-app";
     repo = "immich";
     # Using a fixed commit until upstream has release tags for cli.
-    rev = "014adf175ad50a61f92804666940e267ab329064";
-    hash = "sha256-MK3Watq5/Zp+rymCIfWxAXSgBPDE13g23uDnW7A5x9g=";
+    rev = "8c2195c8205156f6e3168cc52fa34db334568ea9";
+    hash = "sha256-Tseu6aIrYU4Af/jWDi2wDtP77n/aogp7Qkn9mosMaes=";
   };
-
-  npmDepsHash = "sha256-ssxOXKE1t/bSb972w/cBeK61IrqPLmx9ODMn6D+2Ezw=";
-
-  postPatch = ''
-    cd cli
-  '';
-
   meta = {
     description = "CLI utilities for Immich to help upload images and videos";
     homepage = "https://github.com/immich-app/immich";
     license = lib.licenses.mit;
-    mainProgram = "immich";
     maintainers = with lib.maintainers; [ felschr pineapplehunter ];
   };
-}
+
+  open-api-typescript-sdk = buildNpmPackage {
+    pname = "immich-cli-openapi-typescript-sdk";
+    inherit src version meta;
+
+    npmDepsHash = "sha256-WhNdFaFBwb6ehEQgbNJGdzPb3FdJk1Nefi8DcJfY9Wc=";
+
+    postPatch = ''
+      cd open-api/typescript-sdk
+    '';
+  };
+
+  immich-cli = buildNpmPackage {
+    pname = "immich-cli";
+    inherit src version;
+
+    npmDepsHash = "sha256-aSTN+I8B/aLT2ItGoyZTlbdn1VCK0ZmOb1QG7ZQuz+Q=";
+
+    postPatch = ''
+      ln -sv ${open-api-typescript-sdk}/lib/node_modules/@immich/sdk/{build,node_modules} open-api/typescript-sdk
+      cd cli
+    '';
+
+    passthru = {
+      inherit open-api-typescript-sdk;
+      tests.version = testers.testVersion { package = immich-cli; };
+    };
+
+    meta = meta // { mainProgram = "immich"; };
```

Comment:
`meta //` has a tendency to block `builtins.unsafeGetAttrPos "description" package.meta` used by nixos-search, can you rephase it to `meta = { inherit (meta') foo bar baz; mainProgram = "immich"; }`?

Reply:
Thanks for the review!
Fixed.